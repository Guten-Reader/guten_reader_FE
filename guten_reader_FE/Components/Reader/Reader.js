import React from 'react';
import { View, Text, ScrollView, StyleSheet } from 'react-native';
import { createAppContainer } from 'react-navigation';
import { createStackNavigator } from 'react-navigation-stack';
import mockText from '../../mock-data';
import MusicMenu from '../MusicMenu/MusicMenu'

class Reader extends React.Component {
  render() {
    return (
        <ScrollView>
          <View style={styles.container}>
            <Text style={styles.mockText}>{mockText}</Text>
            <MusicMenu />
          </View>
        </ScrollView>
      
    );
  }
}

const AppNavigator = createStackNavigator({
  Reader: {
    screen: Reader
  },
});

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#fff',
    alignItems: 'center',
    justifyContent: 'center',
    marginLeft: '10%',
    marginRight: '10%',
    flex: 1,
  },

  mockText: {
    fontSize: 20,
    marginTop: 40
  }
});

export default createAppContainer(AppNavigator);

