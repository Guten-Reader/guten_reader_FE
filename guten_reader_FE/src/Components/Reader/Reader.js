import React from 'react';
import { View, Text, ScrollView, StyleSheet, Button, StatusBar } from 'react-native';
import { createAppContainer, withNavigation } from 'react-navigation';
import { createStackNavigator } from 'react-navigation-stack';
import * as Font from 'expo-font';
import MusicMenu from '../MusicMenu/MusicMenu';
import GestureRecognizer, {swipeDirections} from 'react-native-swipe-gestures';
import { getToken, getRecommendation, postSongToPlayer, updateCurrentPage } from '../../apiCalls'

class Reader extends React.Component {
  static navigationOptions = {
    title: 'Reader',
    headerTintColor: '#53E69B',
    headerTitleStyle: {
        color: 'black',
      }
    }

  constructor() {g
    super();
    this.state = {
      currentPage: 0,
      currentToken: '',
      currentMood: 0,
      defaultFontSize : 20,
      defaultFontFamily: true,
      isOnDarkMode: false
    };
  };

  async componentDidMount() {
    this.updateCurrentPage()
    const token = await getToken()
    this.setState({ currentToken: token.access_token })
  };

  increaseFontSize() {
    let newFontSize = this.state.defaultFontSize += 2
    this.setState({
      defaultFontSize : newFontSize
    });
 };

 decreaseFontSize() {
   let newFontSize = this.state.defaultFontSize -=2
   this.setState({
     defaultFontSize: newFontSize
   });
 };

  toggleDyslexicFont() {
   if(this.state.defaultFontFamily === true) {
    this.setState({
      defaultFontFamily: false
    });
   } else {
     this.setState({
       defaultFontFamily: true
     });
   };
 };

 toggleDarkMode() {
  if(this.state.isOnDarkMode === false) {
    this.setState({
      isOnDarkMode: true
    });
   } else {
     this.setState({
       isOnDarkMode: false
     });
   };
 };

  async updateCurrentPage() {
    const currentPage = await this.props.navigation.getParam('currentPage', 'ERROR')
    await this.setState({ currentPage: currentPage})
  };

  async onSwipeLeft() {
    this.setState({
      currentPage: this.state.currentPage + 1
    })
    const currentText = await this.props.navigation.getParam('bookText', 'ERROR')
    const recommendation = await getRecommendation(this.state.currentToken, this.state.currentMood, currentText[this.state.currentPage])
    if (recommendation.mood !== undefined) {
      this.setState({ currentMood: recommendation.mood })
    } else {
      this.setState({ currentMood: this.state.currentMood })
    }
    await postSongToPlayer(recommendation.recommended_tracks, this.state.currentToken)
    const bookId = await this.props.navigation.getParam('bookId', 'ERROR')
    await updateCurrentPage(bookId, this.state.currentPage)
  };

  async onSwipeRight() {
    this.setState({
      currentPage: this.state.currentPage - 1
    });
    const bookId = this.props.navigation.getParam('bookId', 'ERROR')
    await updateCurrentPage(bookId, this.state.currentPage)
  };

  render() {
    const bookText = this.props.navigation.getParam('bookText', 'ERROR')
    const bookTitle = this.props.navigation.getParam('title', 'ERROR')
    const currentText = this.props.navigation.getParam('bookText', 'ERROR')
    return (
      <View style={styles.container}>
      <StatusBar barStyle="dark-content" />
        <View style={{backgroundColor: (this.state.isOnDarkMode === true ? 'black' : 'white'), flexDirection: 'row', justifyContent: 'space-between', paddingLeft: 20, paddingRight: 20, paddingTop: 10, paddingBottom: 10}}>
          <Text style={styles.bookTitle}>{bookTitle}</Text>
          <Text style={styles.pageNum}>{this.state.currentPage} / {currentText.length}</Text>
        </View>
        <ScrollView style={{backgroundColor: (this.state.isOnDarkMode === true ? 'black' : 'white'), height: '80%'}}>
          <GestureRecognizer onSwipeLeft={this.onSwipeLeft.bind(this)} onSwipeRight={this.onSwipeRight.bind(this)}>
            <Text style={{
              padding: 20,
              fontSize: (this.state.defaultFontSize),
              fontFamily: (this.state.defaultFontFamily === true ? 'Roboto' : 'OpenDyslexic2'),
              color: (this.state.isOnDarkMode === true ? 'white' : 'black'),
              }}>
              {bookText[this.state.currentPage]}}
            </Text>
          </GestureRecognizer>
        </ScrollView>
        <View style={{backgroundColor: (this.state.isOnDarkMode === true ? 'black' : 'white'), flex: 1, flexDirection: 'row', paddingRight: 20, paddingLeft: 20, justifyContent: 'space-between'}}>
          <Button style={styles.decFont} color="#53E69B" onPress={ this.decreaseFontSize.bind(this)} title="-" titleStyle={{fontSize: 16}} />
          <Button style={styles.incFont} color="#53E69B" onPress={ this.increaseFontSize.bind(this) } title="+" titleStyle={{fontSize: 32}}/>
          <Button color="#53E69B" onPress={ this.toggleDyslexicFont.bind(this) } title={this.state.defaultFontFamily === true ? 'Dyslexic Font' : 'Standard Font'} />
          <Button color="#53E69B" onPress={ this.toggleDarkMode.bind(this) } title={this.state.isOnDarkMode === true ? 'Light Mode' : 'Dark Mode'} />
        </View>
      </View>
    )}
   }

const AppNavigator = createStackNavigator({
  Reader: {
    screen: Reader
  },
});

const styles = StyleSheet.create({
  container: {
    fontFamily: 'OpenDyslexic2',
    flex: 1,
    flexDirection: 'column'
  },
  pageNum: {
    color: '#999999',
    fontWeight: 'bold',
  },
  bookTitle: {
    color: '#999999',
    fontWeight: 'bold',
    width: '75%'
  },
  bookInfo: {
    flexDirection: 'row',
    justifyContent: 'space-between',
    paddingLeft: 20,
    paddingRight: 20,
    paddingTop: 10,
    paddingBottom: 10
  },
  title: {
    color: "black"
  }
});

export default withNavigation(Reader);
