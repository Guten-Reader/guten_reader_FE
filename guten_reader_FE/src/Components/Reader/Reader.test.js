import React from 'react';
// import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';

import Reader from './Reader';

describe('Reader renders correctly with Jest', () => {
  let props;
  let wrapper;

  beforeEach(() => {
    props = {
      currentPage: 0,
      currentToken: '',
      currentMood: 'blank',
      defaultFontSize : 20,
      defaultFontFamily: true,
      isOnDarkMode: false
    };
    wrapper = shallow(<Reader {...props} />);
    mockState = {
      books: [],
      error: ''
    };
  });

  it('renders correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('adds state to the component', () => {
    wrapper.setState({ ...mockState })
    const instance = wrapper.instance()
    expect(instance.state).toEqual(mockState)
  });
});