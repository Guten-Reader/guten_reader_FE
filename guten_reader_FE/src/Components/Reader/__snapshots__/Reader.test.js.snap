// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Reader renders correctly with Jest renders correctly 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;
